cli/lib/appdata.ts

```diff
@@ -151,3 +151,24 @@ export function getNewSitePartial( siteFolder: string ): z.infer< typeof siteSch
 
 	return newSite;
 }
+
+export async function getOrCreateSiteByFolder(
+	siteFolder: string
+): Promise< { site: ReturnType< typeof getNewSitePartial >; userData: UserData } > {
+	const userData = await readAppdata();
+	let site;
+	try {
+		site = await getSiteByFolder( siteFolder );
+	} catch ( error ) {
+		if ( ! ( error instanceof LoggerError ) ) {
+			throw error;
+		}
+		site = getNewSitePartial( siteFolder );
+		if ( ! userData.newSites ) {
+			userData.newSites = [];
+		}
+		userData.newSites.push( site );
+		await saveAppdata( userData );
+	}
+	return { site, userData };
```

Comment:
```suggestion
	return site;
```
Returning `userData` may save us an additional `readAppdata` call in `updateSnapshotInAppdata` and `saveSnapshotToAppdata`, but ultimately, I think it complicates the code by muddying the responsibility of this function.

Let's just add those additional `readAppdata` calls and have this function focused on a single responsibility.